src/renderer/OpenGL.zig

```diff
@@ -1,450 +1,171 @@
-            else
-                null,
-
-            .cursor_opacity = @max(0, @min(1, config.@"cursor-opacity")),
+const configpkg = @import("../config.zig");
+const rendererpkg = @import("../renderer.zig");
+const Renderer = rendererpkg.GenericRenderer(OpenGL);
 
-            .background = config.background.toTerminalRGB(),
-            .foreground = config.foreground.toTerminalRGB(),
-            .invert_selection_fg_bg = config.@"selection-invert-fg-bg",
-            .bold_is_bright = config.@"bold-is-bright",
-            .min_contrast = @floatCast(config.@"minimum-contrast"),
-            .padding_color = config.@"window-padding-color",
+pub const GraphicsAPI = OpenGL;
+pub const Target = @import("opengl/Target.zig");
+pub const Frame = @import("opengl/Frame.zig");
+pub const RenderPass = @import("opengl/RenderPass.zig");
+pub const Pipeline = @import("opengl/Pipeline.zig");
+const bufferpkg = @import("opengl/buffer.zig");
+pub const Buffer = bufferpkg.Buffer;
+pub const Texture = @import("opengl/Texture.zig");
+pub const shaders = @import("opengl/shaders.zig");
 
-            .selection_background = if (config.@"selection-background") |bg|
-                bg.toTerminalRGB()
-            else
-                null,
+pub const cellpkg = @import("opengl/cell.zig");
+pub const imagepkg = @import("opengl/image.zig");
 
-            .selection_foreground = if (config.@"selection-foreground") |bg|
-                bg.toTerminalRGB()
-            else
-                null,
+pub const custom_shader_target: shadertoy.Target = .glsl;
 
-            .custom_shaders = custom_shaders,
-            .links = links,
+/// Because OpenGL's frame completion is always
+/// sync, we have no need for multi-buffering.
+pub const swap_chain_count = 1;
 
-            .arena = arena,
-        };
-    }
+const log = std.log.scoped(.opengl);
 
-    pub fn deinit(self: *DerivedConfig) void {
-        const alloc = self.arena.allocator();
-        self.links.deinit(alloc);
-        self.arena.deinit();
-    }
-};
+/// We require at least OpenGL 4.3
+pub const MIN_VERSION_MAJOR = 4;
+pub const MIN_VERSION_MINOR = 3;
 
-pub fn init(alloc: Allocator, options: renderer.Options) !OpenGL {
-    // Create the initial font shaper
-    var shaper = try font.Shaper.init(alloc, .{
-        .features = options.config.font_features.items,
-    });
-    errdefer shaper.deinit();
+alloc: std.mem.Allocator,
 
-    // For the remainder of the setup we lock our font grid data because
-    // we're reading it.
-    const grid = options.font_grid;
-    grid.lock.lockShared();
-    defer grid.lock.unlockShared();
+/// Alpha blending mode
+blending: configpkg.Config.AlphaBlending,
 
-    var gl_state = try GLState.init(alloc, options.config, grid);
-    errdefer gl_state.deinit();
+/// The most recently presented target, in case we need to present it again.
+last_target: ?Target = null,
 
-    return OpenGL{
+pub fn init(alloc: Allocator, opts: rendererpkg.Options) !OpenGL {
```

Comment:
I think this can trivially have a defined error set too. No need to dive all the way into `pkg/opengl`, can just set it here.